service/history/workflow/context.go

```diff
@@ -724,12 +724,12 @@ func (c *ContextImpl) mergeContinueAsNewReplicationTasks(
 	newRunTask := newWorkflowSnapshot.Tasks[tasks.CategoryReplication][0].(*tasks.HistoryReplicationTask)
 	delete(newWorkflowSnapshot.Tasks, tasks.CategoryReplication)
 
-	newRunBranchToken := newRunTask.BranchToken
+	newRunID := newRunTask.RunID
 	taskUpdated := false
 	for _, replicationTask := range currentWorkflowMutation.Tasks[tasks.CategoryReplication] {
 		if task, ok := replicationTask.(*tasks.HistoryReplicationTask); ok {
 			taskUpdated = true
-			task.NewRunBranchToken = newRunBranchToken
+			task.NewRunID = newRunID
```

Comment:
to support forward / backward compatibility, branch token & run ID should both be set?
task processing logic should use run ID if run ID is set?

Reply:
Good catch. Keep both for compatibility